### ISSUE TITLE
Proposal: Use separate postcss packages instead of postcss-cssnext

### ISSUE THREAD

**Issue by HerrBertling** [xing/hops#169]
I'd like to use separate postcss packages within `hops/packages/config/loaders/postcss.js` instead of `postcss-cssnext`. Currently, we use `cssnext@^2.11.0`. This version uses some older versions for some of the postcss packages included which lead to a strange bug caused by `postcss-nesting@2.3.1` and pseudo elements/media query combinations.

Seeing that https://github.com/MoOx/postcss-cssnext/pull/384 and https://github.com/MoOx/postcss-cssnext/pull/385 seem rather unlikely to be merged soon(ish), I'd propose to replace cssnext with the different plugins bundled in it (excluding the ones deemed unnecessary in https://github.com/MoOx/postcss-cssnext/pull/385) just as standalone ones in the order currently used in cssnext. We could then update all of the packages to the latest versions as planned in the cssnext pull requests.

This removes the "middleman" function (and convenience) of cssnext – but enables us to quickly update the single packages if necessary and thus get the latest fixes per packages on an individual package basis if needs be.

As soon as `cssnext@3.0.0` is released, we can check again and re-introduce it in favor of the single packages approach.

Also, I'd get rid of some superfluous packages (`mixins`, `advanced-variables`, `nested`, `atroot`, `property-lookup`, `extend`) since the functionality is either over the top or implemented within CSS modules etc. in the current setup anyways and can thus be refactored to use already existing functionality.

This would result in:

```js
var options = {browsers: hopsConfig.browsers};

var postcssLoaderOptions = {
  plugins: function (loader) {
    return [
      require('postcss-import')({
        addModulesDirectories: hopsConfig.modules
      }),
      require('postcss-advanced-variables')(),
      require("postcss-custom-properties")(options),
      require("postcss-apply")(options),
      require("postcss-calc")(options),
      require("postcss-image-set-polyfill")(options),
      require("postcss-nesting")(options),
      require("postcss-custom-media")(options),
      require("postcss-media-minmax")(options),
      require("postcss-custom-selectors")(options),
      require("postcss-attribute-case-insensitive")(options),
      require("postcss-color-rebeccapurple")(options),
      require("postcss-color-hwb")(options),
      require("postcss-color-hsl")(options),
      require("postcss-color-rgb")(options),
      require("postcss-color-gray")(options),
      require("postcss-color-hex-alpha")(options),
      require("postcss-color-function")(options),
      require("postcss-font-family-system-ui")(options),
      require("postcss-font-variant")(options),
      require("postcss-initial")(options),
      require("postcss-selector-matches")(options),
      require("postcss-selector-not")(options),
      require("postcss-pseudo-class-any-link")(options),
      require("postcss-replace-overflow-wrap")(options),
      require("autoprefixer")(options),
      require('postcss-reporter')()
    ];
  }
};
```

and would change the `package.json` to look like this – comparison to currently used version shamelessly dumped next to the to-be-used version:

```json
    …
    "postcss": "^6.0.2", // 5.2.17
    "postcss-custom-properties": "^6.0.1", // 5.0.2
    "postcss-apply": "^0.8.0", // 0.3.0
    "postcss-calc": "^6.0.0", // 5.3.1
    "postcss-image-set-polyfill": "^0.3.5", // 0.3.5
    "postcss-nesting": "^4.0.1", // 2.3.1
    "postcss-custom-media": "^6.0.0", // 5.0.1
    "postcss-media-minmax": "^3.0.0", // 2.1.2
    "postcss-custom-selectors": "^4.0.1", // 3.0.0
    "postcss-attribute-case-insensitive": "^2.0.0", // 1.0.1
    "postcss-color-rebeccapurple": "3.0.0", // 2.0.1
    "postcss-color-hwb": "^3.0.0", // 2.0.1
    "postcss-color-hsl": "^2.0.0", // 1.0.5
    "postcss-color-rgb": "^2.0.0", // 1.1.4
    "postcss-color-gray": "^4.0.0", // 3.0.1
    "postcss-color-hex-alpha": "^3.0.0", // 2.0.0
    "postcss-color-function": "^4.0.0", // 2.0.1
    "postcss-font-family-system-ui": "^2.0.1", // 1.0.2
    "postcss-font-variant": "^3.0.0", // 2.0.1
    "postcss-initial": "^2.0.0", // 1.5.3
    "postcss-selector-matches": "^3.0.1", // 2.0.5
    "postcss-selector-not": "^3.0.1", // 2.0.0
    "postcss-pseudo-class-any-link": "^4.0.0", // 1.0.0
    "postcss-replace-overflow-wrap": "^2.0.1-0", // 1.0.0
    "postcss-import": "^10.0.0",
    "postcss-loader": "^2.0.6",
    "postcss-reporter": "^4.0.0",
    …
```

**Comment by dmbch** [comment]
While I really like this proposal, I'll have to think about this a bit: if we go down this route (even temporarily), we lose cssnext's capability of only applying the plugins that are actually needed to support the browsers specified in the config...

**Comment by dmbch** [comment]
@HerrBertling what do you think about using a custom, minimal implementation of cssnext's features? Turns out that that's actually [quite feasible](https://github.com/xing/hops/blob/release/6.0.0-beta.10/packages/config/lib/postcss-necsst.js)...

Let me know what you think. Oh, and of course that thing is called `postcss-necsst` 😁

**Comment by HerrBertling** [comment]
😻

**Comment by dmbch** [comment]
Released in 6.0.0-beta.10
